File: Python/1028-list_sorting.py
from operator import attrgetter


class Student():
    def __init__(self, id, name, grade):
        self.id = id
        self.name = name
        self.grade = grade

    def __str__(self):
        return f'{self.id} {self.name} {self.grade}'


def sort_by_col(students, col=1):
    """sort students with col in-place."""
    if col == 1:
        students.sort(key=attrgetter("id"))
    elif col == 2:
        students.sort(key=attrgetter("name", "id"))
    elif col == 3:
        students.sort(key=attrgetter("grade", "id"))


if __name__ == '__main__':
    n_student, col = [int(i) for i in input().split()]
    students = []
    for _ in range(n_student):
        id, name, grade = input().split()
        students.append(Student(id, name, grade))
    sort_by_col(students, col)
    for student in students:
        print(student)

File: C++/1028-list_sorting.cc
#include <iostream>
#include <vector>
#include <algorithm>
#include <functional>


struct Student {
    std::string id;
    std::string name;
    int grade;
};


bool comp(const Student &stu1, const Student &stu2, const int col) {
    if (col == 2) {
        if (stu1.name < stu2.name) {
            return 1;
        } else if (stu1.name > stu2.name) {
            return 0;
        }
    }
    if (col == 3) {
        if (stu1.grade < stu2.grade) {
            return 1;
        } else if (stu1.grade > stu2.grade) {
            return 0;
        }
    }
    return stu1.id < stu2.id;
}


int main() {

    int n_student, col;
    std::cin >> n_student >> col;

    std::vector<Student> students;
    std::string id, name;
    int grade;
    for (int i=0; i<n_student; ++i) {
        std::cin >> id >> name >> grade;
        students.push_back({id, name, grade});
    }

    using std::placeholders::_1;
    using std::placeholders::_2;
    std::sort(students.begin(), students.end(),
              std::bind(comp, _1, _2, col));
    for (auto student : students) {
        std::cout << student.id << ' '
                  << student.name << ' '
                  << student.grade << '\n';
    }

    return 0;
}
